src/Build/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -473,6 +473,11 @@ public int MaxCPUCount
         /// </summary>
         public bool Question { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value if BuildCheck is enabled. The presence of this flag influences the logging logic.
+        /// </summary>
+        internal bool BuildCheckEnabled { get; set; }
```

Comment:
No need for this to be exposed outside of this class - private field is fine.
Or we can just check `this._componentHost.BuildParameters.IsBuildCheckEnabled` in the logging method